doc-surrealql_versioned_docs/version-latest/statements/delete.mdx

```diff
@@ -19,43 +19,61 @@ DELETE [ ONLY ] @targets
 	[ PARALLEL ]
 ;
 ```
-:::note
-_Note:_ Be aware of this [known issue](/docs/surrealdb/faqs/known-issues#delete-only-always-returns-an-error) when using the `ONLY` clause.
-:::
 
 ## Example usage
 
 ### Basic usage
 
-The following query shows basic usage of the DELETE statement, which is used to delete records from a table or a graph edge.
-
-Deleting records can be done in multiple ways:
+The following queries shows basic usage of the DELETE statement, which is used to delete records from a table or a graph edge.
 
+Deleting records can be done in multiple ways.
 
+Specifying only the table name will delete all the records from a table. Note that a `DELETE` statement returns nothing (i.e. an empty array) by default.
 
 ```surql
 -- Delete all records from a table
 DELETE person;
+```
+
+```bash title="Output"
+[]
+```
+
+A `DELETE` statement on a specific ID will delete a single record.
 
+```surql
 -- Delete a record with a specific numeric id
 DELETE person:100;
 
 -- Delete a record with a specific string id
 DELETE person:tobie;
+```
+
+The `ONLY` keyword can be followed by a `RETURN BEFORE` clause to return just the object for the record in question before it was deleted.
+
+```surql
+DELETE ONLY person:tobie RETURN BEFORE;
+```
+
+Note that as a `DELETE` statement returns an empty array by dafult, and the `ONLY` keyword assumes that a single object will be returned, it will return an error if `RETURN BEFORE` is not included.
```

Comment:
```suggestion
Note that as a `DELETE` statement returns an empty array by default, and the `ONLY` keyword assumes that a single object will be returned, it will return an error if `RETURN BEFORE` is not included.
```